File: labs/lab01/entradaSaida/LabParenteses/lab02.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include<string.h>

// Condição de parada da leitura
bool isFim(char s[]){
    return(strlen(s) == 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}                                                       

//Verificador de palindromo   
bool verifica(char s[]){
    int aux = 0;
    for(int i = 0; i < strlen(s) && aux>=0; i++){
        if(s[i] == '(')aux++;
        if(s[i] == ')')aux--;
    }

    return(aux==0);
}                                   

int main()
{
    char entrada[1000][100];
    int numEntrada = 0;

    // Leitura da entrada padrao
    do{
        scanf(" %[^\n]s", entrada[numEntrada]);
    } while (isFim(entrada[numEntrada++]) == false);
    numEntrada--; 
    
    for (int i = 0; i < numEntrada; i++){
        if (verifica(entrada[i])){
            printf("correto\n"); 
        }
        else{
            printf("incorreto\n");
        }
    }
    return 0;
}       

File: labs/lab01/entradaSaida/LabParenteses/lab02.java
class lab02{

    static boolean verifica(String s){
        int aux = 0;
        for(int i = 0; i < s.length() && aux >=0; i++){
            if(s.charAt(i) == '(')aux++;
            if(s.charAt(i) == ')')aux--;
        }

        return(aux==0);
    }

    public static boolean isFim(String s) {
        return(s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }

    public static void main(String[] args) {
        String[] entrada = new String[1000];
      int numEntrada = 0;

      //Leitura da entrada padrao
      do {
         entrada[numEntrada] = MyIO.readLine();
      } while (isFim(entrada[numEntrada++]) == false);
      numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM

      //Para cada linha de entrada, gerando uma de saida contendo o numero de letras maiusculas da entrada
      for(int i = 0; i < numEntrada; i++){
         if(verifica(entrada[i]) == true){
            MyIO.println("correto");
        } else{
            MyIO.println("incorreto");
        }
      }
    }
}